cognite/powerops/resync/config/_main.py

```diff
@@ -152,7 +152,31 @@ def load_yamls(cls, config_dir_path: Path, instantiate: bool = False) -> dict[st
 
 class CogShopConfig(Config):
     time_series_mappings: Optional[list[TimeSeriesMapping]] = None
-    watercourses_shop: list[ShopFileConfig]
+    watercourses_shop: list[ShopFileConfig]  # validation needs to happen here. If any of the files
+    _dependent_shop_files: set = {
+        "extra_data",
+        "water_value_cut_file_reservoir_mapping",
+        "water_value_cut_file",
+        "module_series",
+    }
+
+    @field_validator("watercourses_shop")
+    @classmethod
+    def validate_shop_related_files(cls, value):
+        """
+        If user has added any of the files above to the configuration, then the "cog_shop_files_config" file
+        must exist among the files to be uploaded
+        """
+        files_to_check = cls._dependent_shop_files.default.intersection({file.cogshop_file_type for file in value})
+        if not files_to_check:
+            return value
+        file_names = [file.file_path.stem for file in value]
+        seen = "cog_shop_files_config" in file_names
```

Comment:
I prefer to have bool variables start with a verb. `seen` is a bit strange name to me
```suggestion
        has_cog_shop_files_config= "cog_shop_files_config" in file_names
```